Source/WebKit/UIProcess/Cocoa/NavigationState.mm

```diff
@@ -420,29 +420,7 @@ static void tryInterceptNavigation(Ref<API::NavigationAction>&& navigationAction
                 return;
             }
 
-            if (!navigationAction->targetFrame()) {
```

Comment:
Why do you think it's safe to remove all of this stuff? It doesn't look like this was added to support AppSSO, so it seems unnecessarily risky to remove this logic without a clear reason.

Reply:
If you think it's just dead code, I would suggest creating a separate PR for that so we can easily revert the change if we discover it breaks something.

Reply:
OK, I'll add this code back and we can remove it as a follow-up if we find out this is fine.

Reply:
Added back the code.